### PR TITLE
Raise a more obvious exception when form data is missing "boundary".

### LINKED ISSUE #1125: Insufficient input validation of content-type 'multipart/form-data'
### Checklist

- [x] The bug is reproducible against the latest release and/or `master`.
- [x] There are no similar issues or pull requests to fix it yet.

### Describe the bug

Not actually a bug, but insufficient input validation.

### To reproduce

Provide POST input (e.g. with postman) with content-type 'multipart/form-data' but without _boundary_ parameter.

### Expected behavior

Shall return 4xx error code. According to rfc7578 boundary is a required parameter.

### Actual behavior

_None_ boundary is passed to multipart library and an exception is raised there.

### Debugging material

```
TypeError: can't concat NoneType to bytes
  File "starlette/middleware/trustedhost.py", line 51, in __call__
    await self.app(scope, receive, send)
  File "starlette/middleware/sessions.py", line 75, in __call__
    await self.app(scope, receive, send_wrapper)
  File "cpms/middleware.py", line 55, in __call__
    await self.app(scope, receive, send)
  File "starlette/middleware/authentication.py", line 48, in __call__
    await self.app(scope, receive, send)
  File "starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "starlette/routing.py", line 582, in __call__
    await route.handle(scope, receive, send)
  File "starlette/routing.py", line 243, in handle
    await self.app(scope, receive, send)
  File "starlette/routing.py", line 54, in app
    response = await func(request)
  File "starlette/authentication.py", line 69, in async_wrapper
    return await func(*args, **kwargs)
  File "cpms/views/files.py", line 90, in decorator
    return await endpoint(request)
  File "cpms/views/files.py", line 133, in upload_files
    for f_name, f_value in (await request.form()).items():
  File "starlette/requests.py", line 240, in form
    self._form = await multipart_parser.parse()
  File "starlette/formparsers.py", line 181, in parse
    parser = multipart.MultipartParser(boundary, callbacks)
  File "multipart/multipart.py", line 1042, in __init__
    self.boundary = b'\r\n--' + boundary
```

### Environment

- OS: OpenBSD 6.7
- Python version: 3.7.9
- Starlette version: 0.14.1

### Additional context

The problem occurs due insufficient checking of the input on line 166 of starlette/formparsers.py
`      boundary = params.get(b"boundary")`

It is better to replace it with:
`      boundary = params[b"boundary"]`
so exception at early stage will be raised.

Or make an additional input validation.
```
      boundary = params.get(b"boundary")
      if not boundary:
          raise HTTPException(status_code=400)
```

### PR DESCRIPTION
**Pull request by toxadx**
The best strategy is to let KeyError raise when parsing request with missing boundary parameter, so end-user can process KeyError exception in their code.

Closes #1125
Related to #1303

### CODE REVIEW

File: tests/test_formparsers.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import os
 import typing
 
+import pytest
```
Comment (lovelydinosaur) [inline tests/test_formparsers.py:4]:
```suggestion
import pytest

```

File: tests/test_formparsers.py [hunk 2]
```diff
@@ -386,3 +388,20 @@ def test_user_safe_decode_helper():
 def test_user_safe_decode_ignores_wrong_charset():
     result = _user_safe_decode(b"abc", "latin-8")
     assert result == "abc"
+
+
+def test_missing_boundary_parameter(test_client_factory):
+    client = test_client_factory(app)
+    with pytest.raises(KeyError, match='boundary') as exc:
```
Comment (lovelydinosaur) [inline tests/test_formparsers.py]:
```suggestion
    with pytest.raises(KeyError, match="boundary") as exc:
```

File: tests/test_formparsers.py [hunk 3]
```diff
@@ -386,3 +388,20 @@ def test_user_safe_decode_helper():
...
+            "/",
+            data=(
+                # file
+                b'Content-Disposition: form-data; name="file"; filename="\xe6\x96\x87\xe6\x9b\xb8.txt"\r\n'  # noqa: E501
+                b"Content-Type: text/plain\r\n\r\n"
+                b"<file content>\r\n"
+            ),
+            headers={
+                "Content-Type": "multipart/form-data; charset=utf-8"
+            },
```
Comment (lovelydinosaur) [inline tests/test_formparsers.py]:
```suggestion
            headers={"Content-Type": "multipart/form-data; charset=utf-8"},
```

File: tests/test_formparsers.py [hunk 4]
```diff
@@ -386,3 +388,18 @@ def test_user_safe_decode_helper():
 def test_user_safe_decode_ignores_wrong_charset():
     result = _user_safe_decode(b"abc", "latin-8")
     assert result == "abc"
+
+
+def test_missing_boundary_parameter(test_client_factory):
+    client = test_client_factory(app)
+    with pytest.raises(KeyError, match="boundary") as exc:
```
Comment (lovelydinosaur) [inline tests/test_formparsers.py]:
```suggestion
    with pytest.raises(KeyError, match="boundary"):
```

**Review by lovelydinosaur — APPROVED** [review]
Raising a `KeyError` here isn't perfect, but I'd agree that it's *better*.
So yup, let's go with this. Thanks!

**Comment by Kludex** [comment]
We should probably catch that and raise something else from Starlette itself :thinking:

**Comment by lovelydinosaur** [comment]
@Kludex - I'd be okay with that if someone's sufficiently motivated, yup.

(Rolled with this for now, since as @toxadx mentions, this *is* at least an improvement for now.)

**Pull request merged**